e2e/runner_test.go

```diff
@@ -3,15 +3,77 @@ package e2e
 import (
 	"flag"
 	"os"
+	"path"
 	"strconv"
 	"testing"
 
 	"github.com/cucumber/godog"
 	"github.com/cucumber/godog/colors"
+	log "github.com/sirupsen/logrus"
 
 	"github.com/elastic/metricbeat-tests-poc/cli/config"
 )
 
+type contextMetadata struct {
+	name         string
+	modules      []string
+	contextFuncs []func(s *godog.Suite) // the functions that hold the steps for a specific
+}
+
+func (c *contextMetadata) getFeaturePaths() []string {
+	paths := []string{}
+	for _, module := range c.modules {
+		paths = append(paths, path.Join("features", module, c.name+".feature"))
+	}
+
+	return paths
+}
+
+var supportedProducts = map[string]*contextMetadata{
```

Comment:
this structure can grow a lot, and decouple test from categories, it is odd for me.

Reply:
Yeah, same for me :( I think this is a workaround until I find how to fix it in the upstream project (as explained here https://github.com/mdelapenya/sample-godog)